concordium-consensus/src/Concordium/GlobalState.hs

```diff
@@ -391,12 +397,13 @@ class GlobalStateConfig (c :: Type) where
     -- storage. This may throw a 'GlobalStateInitException'.
     --
     -- The return value is 'Nothing' if the state could not be found. If the
-    -- state could be found, but could not be loaded an exception is thrown.
+    -- state could be found, but could not be loaded, recovery is attempted,
+    -- and if that fails,  an exception is thrown.
     --
     -- Note that even if the state is successfully loaded it is not in a usable
     -- state for an active consensus and must be activated before. Use
     -- 'activateGlobalState' for that.
-    initialiseExistingGlobalState :: forall pv . IsProtocolVersion pv => SProtocolVersion pv -> c -> LogIO (Maybe (GSContext c pv, GSState c pv))
+    initialiseExistingGlobalState :: forall pv . IsProtocolVersion pv => SProtocolVersion pv -> Either ByteString PVGenesisData -> c -> LogIO (Maybe (GSContext c pv, GSState c pv))
```

Comment:
I think a simpler way of handling this would be to return `Nothing` if only one of the treestate or blockstate is found, and then have `initialiseNewGlobalState` remove any treestate/blockstate that already exists.

Reply:
Yes, that's what we decided upon yesterday.